captum/attr/_core/modified_relu_grad.py

```diff
@@ -18,11 +18,72 @@ def __init__(self, model):
         super().__init__(model)
         self.model = model
         self.backward_hooks = []
+        self.relu_grad_output = relu_grad_output
         assert isinstance(self.model, torch.nn.Module), (
             "Given model must be an instance of torch.nn.Module to properly hook"
             " ReLU layers."
         )
 
+    def attribute(self, inputs, target=None, additional_forward_args=None):
+        r""""
+        Computes attribution by overriding relu gradients. Based on constructor
+        flag relu_grad_output, performs either GuidedBackpropagation if False
+        and Deconvolution if True. This class is the parent class of both these
+        methods, more information on usage can be found in the docstrings for each
+        implementing class.
+        """
+
+        # Keeps track whether original input is a tuple or not before
+        # converting it into a tuple.
+        is_inputs_tuple = isinstance(inputs, tuple)
+
+        inputs = format_input(inputs)
+        gradient_mask = apply_gradient_requirements(inputs)
+
+        # set hooks for overriding ReLU gradients
+        warnings.warn(
+            "Setting backward hooks on ReLU activations."
+            "The hooks will be removed after the attribution is finished"
+        )
+
+        self.model.apply(self._register_hooks)
+
+        gradients = self.gradient_func(
+            self.forward_func, inputs, target, additional_forward_args
+        )
+
+        # remove set hooks
+        self._remove_hooks()
+
+        undo_gradient_requirements(inputs, gradient_mask)
+        return _format_attributions(is_inputs_tuple, gradients)
+
+    def _register_hooks(self, module):
+        if isinstance(module, torch.nn.ReLU):
+            hook = module.register_backward_hook(self._backward_hook)
+            self.backward_hooks.append(hook)
+
+    def _backward_hook(self, module, grad_input, grad_output):
+        to_override = grad_output if self.relu_grad_output else grad_input
+        if isinstance(to_override, tuple):
+            return tuple(F.relu(tensor) for tensor in to_override)
```

Comment:
nit: perhaps we can call it `tuple(F.relu(to_override_grad) for tensor in to_override_grads)`, because tensor is a keyword and a generic term